packages/fast-components-react-msft/src/progress/progress.tsx

```diff
@@ -47,48 +48,134 @@ class Progress extends Foundation<ProgressHandledProps, ProgressUnhandledProps,
-                            this.props,
-                            "managedClasses.progress_valueIndicator"
-                        )}
-                        style={{ width: `${this.props.value}%` }}
-                    />
-                </div>
-                <div
-                    slot={ProgressType.indeterminate}
-                    className={get(this.props, "managedClasses.progress_indicator")}
-                >
-                    {this.renderIndeterminateItems()}
-                </div>
+                {this.renderProgress()}
             </BaseProgress>
         );
     }
 
+    /**
+     * Create class names
+     */
+    protected generateClassNames(): string {
+        let className: string = get(this.props.managedClasses, "progress", "");
+
+        if (this.props.circular) {
+            className = get(this.props.managedClasses, "progress_circular", "");
+        }
+
+        return super.generateClassNames(className);
+    }
+
+    private generateCircularValueIndicatorClassNames(): string {
+        let className: string = get(
+            this.props.managedClasses,
+            "progress_valueIndicator",
+            ""
+        );
+
+        if (!this.props.value) {
+            className = `${className} ${get(
+                this.props.managedClasses,
+                "progress_valueIndicator__indeterminate",
+                ""
+            )}`;
+        }
+
+        return className;
+    }
+
+    private generateSVGClassNames(): string {
+        const className: string = this.props.size
+            ? get(this.props.managedClasses, `progress_circular__${this.props.size}`, "")
+            : get(this.props.managedClasses, "progress_circular__container", "");
+
+        return className;
+    }
+
     private progressIndicatorClasses(): string {
         return [
-            get(this.props, "managedClasses.progress_indicator"),
-            get(this.props, "managedClasses.progress_indicator__determinate"),
+            get(this.props.managedClasses, "progress_indicator"),
+            get(this.props.managedClasses, "progress_indicator__determinate"),
         ].join(" ");
     }
 
     private renderIndeterminateItems(): JSX.Element[] {
         return new Array(Progress.indicatorCount)
             .fill(undefined)
             .map((item: undefined, index: number) => {
-                let className: string = get(this.props, "managedClasses.progress_dot");
+                let className: string = get(this.props.managedClasses, "progress_dot");
                 className = `${className} ${get(
-                    this.props,
-                    `managedClasses.progress_dot__${index + 1}`
+                    this.props.managedClasses,
+                    `progress_dot__${index + 1}`
                 )}`;
 
                 return <span className={className} key={index} />;
             });
     }
+
+    private renderCircularBackground(): JSX.Element {
+        return (
+            <circle

```

Comment:
This same circle is created three times - maybe abstract to a small functional component that can accept className and style?